packages/push/src/PushRegistration.ts

```diff
@@ -96,22 +97,102 @@ export class PushRegistration {
       "categories": categories
     };
 
-    const instance = axios.create({
+    const http = axios.create({
       baseURL: url,
       timeout: 5000,
       headers: {"Authorization": `Basic ${authToken}`}
     });
 
     return new Promise((resolve, reject) => {
-      return instance.post(PushRegistration.API_PATH, postData)
+      return http.post(PushRegistration.API_PATH, postData)
       .then(
         () => {
+          const storage = window.localStorage;
+          storage.setItem(PushRegistration.TOKEN_KEY, deviceToken);
+
           if (isCordovaAndroid()) {
             this.subscribeToFirebaseTopic(variantId);
             for (const category of categories) {
               this.subscribeToFirebaseTopic(category);
             }
           }
+
+          resolve();
+        }
+      )
+      .catch(reject);
+    });
+  }
+
+  /**
+   * Unregister device for Android or IOS platforms
+   */
+  public unregister(): Promise<void> {
+    if (!window || !window.device) {
+      return Promise.reject(new Error("Registration requires cordova plugin. Verify the " +
+        "@aerogear/cordova-plugin-aerogear-push plugin is installed."));
+    }
+
+    if (!this.pushConfig || !this.pushConfig.config || !this.pushConfig.url) {
```

Comment:
@danielpassos I see this code repeated here and in the `register` function. To me all this is doing is validating some configs that are actually passed into the constructor. Why not move all these checks there? Additionally, we're doing this `const http axios.create(...)` passing in some stuff that is all available from the configs passed into the constructor. I think it makes a lot of sense to do the following

* validate all of this config once in the constructor but do not throw (because I'm guessing we would break existing apps (showcase) where UPS is not always there). Set something like `this.validationError = <error message>`
* If no validation errors are present then set `this.httpClient = axios.create(...)` within the constructors
* then within `register` and `unregister` you can remove several lines of code and have a much more simple flow:

```
function register(...) : Promise<void> {
  if (this.validationError) {
    return Promise.reject(this.validationError)
  }

  ...
  this.httpClient.post(...) //or this.httpClient.delete()
}
```

That would really simplify this file and make it a lot more maintainable IMO. WDYT?